pkgs/development/compilers/gcc/patches/default.nix

```diff
@@ -156,15 +156,21 @@ in
   }) ];
   "12" = [ (fetchurl {
     name = "gcc-12-darwin-aarch64-support.patch";
-    url = "https://raw.githubusercontent.com/Homebrew/formula-patches/f1188b90d610e2ed170b22512ff7435ba5c891e2/gcc/gcc-12.3.0.diff";
-    sha256 = "sha256-naL5ZNiurqfDBiPSU8PTbTmLqj25B+vjjiqc4fAFgYs=";
+    url = "https://raw.githubusercontent.com/Homebrew/formula-patches/1ed9eaea059f1677d27382c62f21462b476b37fe/gcc/gcc-12.4.0.diff";
+    sha256 = "sha256-wOjpT79lps4TKG5/E761odhLGCphBIkCbOPiQg/D1Fw=";
   }) ];
   "11" = [ (fetchpatch {
     # There are no upstream release tags in https://github.com/iains/gcc-11-branch.
     # ff4bf32 is the commit from https://github.com/gcc-mirror/gcc/releases/tag/releases%2Fgcc-11.4.0
     url = "https://github.com/iains/gcc-11-branch/compare/ff4bf326d03e750a8d4905ea49425fe7d15a04b8..gcc-11.4-darwin-r0.diff";
     hash = "sha256-6prPgR2eGVJs7vKd6iM1eZsEPCD1ShzLns2Z+29vlt4=";
   }) ];
+  "10" = [ (fetchpatch {
+      # There are no upstream release tags in https://github.com/iains/gcc-10-branch.
+      # ff4bf32 is the commit from https://github.com/gcc-mirror/gcc/releases/tag/releases%2Fgcc-10.5.0
+      url = "https://github.com/iains/gcc-10-branch/compare/d04fe5541c53cb16d1ca5c80da044b4c7633dbc6...gcc-10-5Dr0-pre-0.diff";
```

Comment:
It would really be best to use a commit hash for the endpoints here, I think. Not a new problem as of this PR though, so not a blocker.

Reply:
Ah, sorry, i missed the other discussion. Does `..` rather than `...` work?

Reply:
Unfortunately, no. Both seem to do the same thing with a commit. I’ll fix the commit hash in the comment though.

Once a release is done for 10.5-darwin (since this seems to be a pre-release branch on top of the 10.5 release), I can open an issue to see if it can be tagged.

Reply:
Fair enough. It might be better to just vendor the patch, but if it’s unlikely there’ll be any more commits then it might not be worth it.

Reply:
The patch is 1 MiB.

Reply:
Probably? Anyway I think this is fine; if it causes problems later we can revisit.